### ISSUE TITLE
Broken on 30.10.22 when passing from sommer to winter time

### ISSUE THREAD

**Issue by kevinleto-informaticon** [ANovokmet/svelte-gantt#113]
Hi guys, I've noticed that the Gantt doesn't work fine from 30.10.22 when we switched from the summertime to the winter one (-1 hour).

The calendar shows two times the 30th instead of the 31st + a small portion of the 1st of November.
![image](https://user-images.githubusercontent.com/62250375/199682058-bed1a6af-6aa8-48d3-8e15-86ff077454df.png)

This causes all tasks after the 30th, to be either not displayed or wrongly positioned on the row.
Here is how I set the time range in Gantt:

```ts
let october30: Moment = moment("2022-10-30", "YYYY-MM-DD").startOf('day');
let october31: Moment = moment("2022-10-31", "YYYY-MM-DD").endOf('day');
let dayCount = Math.abs(october31.diff(october30, 'day')) + 1;

gantt.$set({
  from: october30.valueOf(),
  to: october31.valueOf(),
  minWidth: dayCount * 20
});
```

I would expect that the Gantt shows the 30th and the 31st completely (without the portion of the 1st of November).

Can you please help me to solve the problem?
Thanks a lot for the help 🙏 

PS: this problem occurs each time we pass from summertime to winter one (on 28 March 2021 I become the same problem). Not sure if it also happens in the opposite direction. I've started using the library some months ago.
I use the new version of the lib `"svelte-gantt": "~4.0.9-beta"`

**Comment by kevinleto-informaticon** [comment]
Something similar happens also in the demo.
When setting the dates in the Nav component, Gantt shows two times the 30th and then goes ahead with 31st, 1st, 2nd, and so on.
![image](https://user-images.githubusercontent.com/62250375/199697782-6c1e7e78-215e-486f-b7b0-8bb40f21afb3.png)

I've tested it with the `v4.0.7-beta`, as the new version cannot be started [see this issue](https://github.com/ANovokmet/svelte-gantt/issues/107).

**Comment by kevinleto-informaticon** [comment]
Edit:
I think the error is [here](https://github.com/ANovokmet/svelte-gantt/blob/d68cd1a2e5f6e547c62693f0eb181191955b2d0a/src/utils/date.ts#L169) as the 30th has a duration of 25 (24+1) hours and not 24.

The bug is **not** present in the `v3.0.4-beta`.

**Comment by V-Py** [comment]
Will look at it asap, but if you find any corrections you might as well do a PR so I can integrate it way faster

**Comment by V-Py** [comment]
Should be fixed since #136
